ansible_rulebook/event_source/webhook.py

```
@@ -105,11 +147,153 @@
+                async with session.post(
+                    self.introspection_url, data=data
+                ) as response:
+                    response.raise_for_status()
+                    response_data = await response.json()
+                    if not response_data.get("active", False):
+                        message = "User is not active"
+                        logger.warning(message)
+                        raise AuthenticationFailed(message)
+            except aiohttp.ClientResponseError as err:
+                logger.error(f"HTTP Error {err.status}: {err.message}")
+                raise AuthenticationFailed(err.message)
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)